fido2.go

```diff
@@ -234,56 +238,101 @@ func DeviceLocations() ([]*DeviceLocation, error) {
 }
 
 // NewDevice opens device at path.
+// Opened devices must be explicitly closed.
 func NewDevice(path string) (*Device, error) {
 	if path == "" {
 		return nil, errors.Errorf("empty device path")
 	}
+
+	dev, err := openDevice(path)
+	if err != nil {
+		return nil, fmt.Errorf("open device: %w", err)
+	}
+
 	return &Device{
-		path: fmt.Sprintf("%s", path),
+		path: path,
+		dev:  dev,
 	}, nil
 }
 
-func (d *Device) open() (*C.fido_dev_t, error) {
+func openDevice(path string) (*C.fido_dev_t, error) {
+	pathC := C.CString(path)
+	defer C.free(unsafe.Pointer(pathC))
```

Comment:
Note that there are many instances of CString and CBytes being leaked in the existing upstream code (see the body of Assertion and MakeCredential methods). I'm not fixing them here.

See CString and CBytes here: https://pkg.go.dev/cmd/cgo#hdr-Go_references_to_C.

Reply:
Done. Good call.

Reply:
On a second look it looks like I was wrong. For example: https://github.com/Yubico/libfido2/blob/f03f4c4d9d25e932066bef5f26b8b82a686ccd33/src/assert.c#L824.

I'll ditch the comment.